src/include/rbd/librbd.hpp

```diff
@@ -38,6 +39,7 @@ namespace librbd {
     uint64_t id;
     uint64_t size;
     std::string name;
+    utime_t stamp;
```

Comment:
The only real solution would be to create a new ```snap_get_timestamp``` API method to retrieve the timestamp by snapshot name.